### ISSUE TITLE
`compile_error` macro does not emit diagnostic on top level

### ISSUE THREAD

**Issue by Veykril** [rust-analyzer/rust-analyzer#8459]
```rust
compile_error!("top level"); // < doesn't emit anything
fn main() {
    compile_error!("non top level");
    // ^^^^ emits `non top level` diagnostic
}
const _: () = compile_error!("non top level"); // < doesn't emit anything
```
![image](https://user-images.githubusercontent.com/3757771/114277913-875a9b00-9a2d-11eb-95a9-e80a5aea7095.png)
